Stargazing search indexing via firebase

camallen opened the pull request.

camallen committed:
only skip flagged==true comments

camallen committed:
add firebase authentication to the browser client

camallen committed:
use stargazing url

camallen committed:
bump libs

camallen committed:
dont run the example app to query ES

camallen committed:
index everything, front end will take care of flagging comments

camallen committed:
only write data to the response once

adammcmaster commented on start_firebase_listener.sh:
```diff
@@ -40,6 +40,8 @@ fi
-node app.js &
-cd example
-serve
+node app.js
```
Didn't notice this yesterday, but it's probably a good idea to prefix this command with exec (http://wiki.bash-hackers.org/commands/builtin/exec)

camallen replied:
will do.

camallen committed:
cleanup search queue code

camallen committed:
wait for a response message with a total not 0

Not sure why but the .on("value") event is being called when the ES results are written to firebase. Seems the first response is { total: 0 }

camallen committed:
use exec to run the shell command

camallen committed:
serialize the whole search response for the client

camallen committed:
respond only to full search results not just the hits

camallen committed:
do not detach with exec
